---
title: "root_stem_crown_comparison"
output: html_notebook
author: 'Aaron Yerke'
---

This the R Notebook for collaboration with Dr. Yogini Jaiswal for her comparison of roots, stems, and crowns.

From Yogini:
I. Changes done in all files and annotations
* In all the files refer to the sheet: “Combined stem and roots” and “For PCA – Stem and Roots”
* All cells in pink indicate that the constituent is not present in the sample
* I believe in the previous study, we used “zeros” in rows where samples were not present.
* All values present in “Combined stem and roots” are normalised with internal standard area for GC-MS File.
* The data generated for analytes represent relative rather than absolute concentrations (semi-quantitative) based on their areas. All values generated for these compounds are therefore “unitless” but strongly correlate to absolute concentration when comparing values for the same metabolite; however, this does not apply when comparing values between different metabos.
* For LC-MS analysis there are two files, negative and positive representing 2 different modes of ionisation used for analysis to have a more comprehensive identification of compounds that can be detected/ionised in both positive and negative mode. 
* There can be compounds that may be common between both ‘positive’ and ‘negative’ mode of LC-MS analysis files. 

II. Nomenclature:
A. Crown root parts: 
  CRP: Crown root peel. 
  CNP: Crown root with ‘no’ peel
  CWP: Whole crown root
B. Lateral root parts: 
  LRP: whole lateral root
C. Stem parts: 
  SWP: Whole stem. 
  SNP: Stem with ‘no’ peel
  SP: Stem peel

III.A  Suggested comparisons by Dr. Jaiswal with Aaron's Notes
1. Whole crown root / lateral root/ stem
  Cycle through all of the metabolites and do ALL the metadata catagories
    lme(metabo ~ metad, method= "REML", random = ~1 | plant_number, data = myD)
    then anova
    
    table showing pvals for each test for each metabolite
      metabo test1 test2 test3 ...
    
2. Each part of Crown root and lateral root.
  lme with random effects of sample number
  
3. Each part of Crown root
  3 Samples each with 3 sections: CRP, CNP, CWO
    lme with random effect on sample number
    ANOVA
    
III.B  Additional tests suggested by Aaron
1. Comparison of each metabolite to the top three mds axis
  Add Pval for plant_num to show this isn't a batch effect

2. Comparison of each metadata to the top three mds axis


IV. Changes done in files for PCA analysis:

File 1: NCAT155_GCMS_7July2017 (1)
1. Refer to the sheet: “Combined stem and roots” and “For PCA – Stem and Roots”. 
2. Removed the internal standard row - “Docosanoic acid, methyl ester (IS)”, as it is an internal standard used for quality control and not a constituent. 
3. Removed putative annotations for “present” and “not present”

File 2: NCAT155-Negative mode - LC-MS
1. Refer to the sheet: “Combined stem and roots” and “For PCA – Stem and Roots”. 
2. In this file, all samples marked in green, are the same constituent but different mass adducts. 
    So I have discarded them in the sheet “For PCA – Stem and Roots”.  

File 3: NCAT155-Positive mode- LC-MS-SORTED- Roots
1. Refer to the sheet: “Combined stem and roots” and “For PCA – Stem and Roots”. 
2. In this file, all samples marked in green, are the same constituent but different mass adducts. 
    So I have discarded them in the sheet “For PCA – Stem and Roots”.  
3. For compounds with two adducts, the adduct rows with lower areas are deleted (will check with you before deletion)

Bioactive compounds:
  Kavain
  Dihydromethysticin
  Desmethoxyyangonin

Metadata categories for publication:
  tissue
  plant part
  plant number

First load the data:
```{r}
rm(list = ls()) #clear workspace

#declare dataset
ds = 'gc'

#read in meta-data
metadata = read.table(file.path('.','gc','meta_data.csv'), 
  sep=",", 
  header=TRUE, 
  row.names = 1, 
  check.names = FALSE,
  stringsAsFactors=FALSE)

#read in raw numeric data
metabolites = read.table(file.path('.', ds,'gc_data_numeric.csv'), 
  sep=",", 
  header=TRUE, 
  row.names = 1, 
  check.names = FALSE)

if (!require("RColorBrewer")) {
install.packages("RColorBrewer")
library(RColorBrewer)
}

#set color palette
palette( brewer.pal(7,"Accent") )

dim(metabolites)

```
```{r}
#boxplot before transform
boxplot(metabolites,
  main = paste(toupper(ds), 'Data without log-transform'),
  col = as.factor(metadata['tissue',]),
  xlab = "tissues",
  xaxt = 'n'
)
mtext(text= unique(as.factor(metadata['tissue',])),
  side = 1,
  outer = FALSE,
  at = seq(2, 21, by = 3),
  line = 1
  )
par(new=TRUE)
stripchart(metabolites,
  method='jitter',
  jitter=.2,
  vertical=TRUE,
  add=T,
  pch=16,
  cex = 0.6,
  col = 'grey'
  )
plot.new()
legend('bottomright', 
  legend = unique(as.factor(metadata['tissue',])), 
  col = unique(as.factor(metadata['tissue',])), 
  pch = 19,
  cex=0.4
  )
```
This plot is hard to see.  Let's try log transform..


```{r}
metabolites = log2(metabolites + 1)
```

```{r}
par(bty = 'l', 
    mar = c(5, 4, 5, 2) + 0.1)
boxplot(metabolites,
  main = paste(toupper(ds), 'Data with log-transform'),
  col = as.factor(metadata['tissue',]),
  xlab = "tissues",
  xaxt = 'n'
)
mtext(text= unique(as.factor(metadata['tissue',])),
  side = 1,
  outer = FALSE,
  at = seq(2, 21, by = 3),
  line = 1
  )
par(new=TRUE)
stripchart(metabolites,
  method='jitter',
  jitter=.2,
  vertical=TRUE,
  add=T,
  pch=16,
  cex = 0.6,
  col = 'grey'
  )
plot.new()
legend('bottomright', 
  legend = unique(as.factor(metadata['tissue',])), 
  col = unique(as.factor(metadata['tissue',])), 
  pch = 19,
  cex=0.4
  )
```
This tells me that we need to use the log transform.


To make publication size pdf:
```{r}
pdf(paste(ds, "OverallBoxplot.pdf", sep=""), width = 5, height = 4)
  par(bty = 'l', 
    mar = c(5, 4, 5, 2) + 0.1)
boxplot(metabolites,
  main = paste(toupper(ds), 'Data with log-transform'),
  col = as.factor(metadata['tissue',]),
  xlab = "tissues",
  xaxt = 'n'
)
#  axis(side = 2, seq(14,26,2)) figure this out later
mtext(text= unique(as.factor(metadata['tissue',])),
  side = 1,
  outer = FALSE,
  at = seq(2, 21, by = 3),
  line = 1
  )
par(new=TRUE)
stripchart(metabolites,
  method='jitter',
  jitter=.2,
  vertical=TRUE,
  add=T,
  pch=16,
  cex = 0.6,
  col = 'grey'
  )
plot.new()
legend('bottomright', 
  legend = unique(as.factor(metadata['tissue',])), 
  col = unique(as.factor(metadata['tissue',])), 
  pch = 19,
  cex=0.4
  )
dev.off()
```


Next look at PCA.
```{r}
#create PCA
group_pca = prcomp(t(na.omit(metabolites)), 
  center = TRUE,
  scale = TRUE)

#extract PCA matrix and convert to dataframe
myPCA = data.frame(group_pca$x)

summary(group_pca)

```
PCA1 and PCA2 account for 84% cummaltively.


PCA1 and PCA2
```{r}
for (i in 1:nrow(metadata)){
  par(bty = 'l', 
    mar = c(5, 4, 5, 2) + 0.1,
    cex.lab = 1.25
    )
  plot(myPCA$PC1, myPCA$PC2,
    col='black',
    pch = 21,
    bg = as.factor(metadata[i,]),
    cex = 3
  )
  plot.new()
  legend('topright', legend = unique(as.factor(metadata[i,])),
    col = unique(as.factor(unique(metadata[i,]))),
    pch = 19,
    cex=0.6
    )
}
```
PCA1 and PCA2 -> file for sending to collaborators
```{r}
plot2PcasPdf = function(pcaA, pcaB, xl, yl, pdfName){
  pdf(paste(ds, pdfName, sep=""))
  for (i in 1:nrow(metadata)){
    par(bty = 'l', 
      mar = c(5, 4, 5, 2) + 0.1,
      cex.lab = 1.25
      )
    plot(pcaA, pcaB,
      col='black',
      pch = 21,
      bg = as.factor(metadata[i,]),
      cex = 3,
      main = paste(ds, 'dataset\n', 'Category: ', row.names(metadata)[i]),
      xlab = xl,
      ylab = yl
    )
    plot.new()
    legend('topright', legend = unique(as.factor(metadata[i,])),
      col = unique(as.factor(unique(metadata[i,]))),
      pch = 19,
      cex=0.6
      )
  }
  dev.off()
  print('plot created')
}
plot2PcasPdf(myPCA$PC1,
             myPCA$PC2,
             'MDS 1, 66% of Variance',
             'MDS 2, 17% of Variance',
             "Pca1_2.pdf")
```

PCA1 and PCA3
```{r}
for (i in 1:nrow(metadata)){
  par(bty = 'l', 
    mar = c(5, 4, 5, 2) + 0.1)
  plot(myPCA$PC1, myPCA$PC3,
    xlab = 'MDS 1, 66% of Variance',
    ylab = 'MDS 3, 3% of Variance',
    col=as.factor(metadata[i,]),
    main = paste(ds, 'dataset\n', row.names(metadata)[i]),
    sub = paste(ds, 'dataset'),
    pch = 19,
    cex = 1.3
    )
  legend('bottomright', legend = unique(as.factor(metadata[i,])), 
    col = unique(as.factor(unique(metadata[i,]))), 
    pch = 1, 
    cex=0.6)
}
```

PCA1 and PCA3: saving to pdf
```{r}
plot2PcasPdf(myPCA$PC1, 
             myPCA$PC3,
             'MDS 1, 66% of Variance',
             'MDS 3, 3% of Variance', 
             "Pca1_3.pdf")
```



PCA2 and PCA3
```{r}
for (i in 1:nrow(metadata)){
  par(bty = 'l', 
    mar = c(5, 4, 5, 2) + 0.1)
  plot(myPCA$PC2, myPCA$PC3,
    col=as.factor(metadata[i,]),
    main = paste(ds, 'dataset\n', row.names(metadata)[i]),
    xlab = 'MDS 2, 17% of Variance',
    ylab = 'MDS 3, 6% of Variance',
    pch = 19,
    cex = 1.3
    )
  legend('bottomright', legend = unique(as.factor(metadata[i,])), 
    col = unique(as.factor(unique(metadata[i,]))),
    pch = 1,
    cex=0.6
  )
}
```

Make PDF of PCA2 and PCA3
```{r}
plot2PcasPdf(myPCA$PC2, 
             myPCA$PC3,
             'MDS 2, 17% of Variance',
             'MDS 3, 6% of Variance',
             "Pca2_3.pdf")
```



Compare each PCA or metabolite to each metadata Category using lm and anova.

Expected output: Table, 
  columns: PCAs, metadata Category, pvals
```{r}
lmMetadata = function(testingData, label){
  
  index = c()
  metaName = c()
  metaPval = c()
  indexName = c()
  r_sqrd = c()
  
  for (i1 in 1:nrow(testingData)){
    for (i2 in 1:nrow(metadata)){
      tryCatch(
       { 
        myLm = lm( unlist(testingData[i1,]) ~ as.factor(as.character(metadata[i2,])))
        index = c(index, i1)
        metaName = c(metaName, row.names(metadata)[i2])
        indexName = c(indexName, row.names(testingData)[i1])
        metaPval =  c(metaPval, anova(myLm)$"Pr(>F)"[1])
        r_sqrd =  c(r_sqrd, summary(myLm)$r.squared)
        },
        error=function(cond) {
          print('an error is thrown')
          message(cond)
        },
        warning=function(cond) {
          print('a warning is thrown')
          message(cond)
          # Choose a return value in case of warning
          #return(NULL)
        }
      )
    }
  }
  
  pdf(paste(ds, "HistogramRsqrdMetadata", label, "Lm.pdf", sep=""))
  hist(r_sqrd, 
    main = paste(ds, ' Histogram of lm r squared ', label),
    xlab = 'Adjusted R squared',
    breaks = 50
  )
  dev.off()
  
  pdf(paste(ds, "HistogramPvalMetaVs", label, "Lm.pdf", sep=""))
  hist(metaPval, 
    main = paste(ds, ' Histogram of lm pvalues for ', label),
    xlab = 'Uncorrected pvalues',
    breaks = 50
  )
  dev.off()
  
  metaPvalAdj = p.adjust(metaPval, method = "BH")
  dFrame <- data.frame(metaPvalAdj,metaName,index, indexName)
  dFrame <- dFrame [order(dFrame$metaPvalAdj),]
  
  write.table(dFrame, file=paste(ds, "MetaVs",label,"LinearMod.txt", sep = ''), row.names=FALSE, sep="\t")
  
  write.table(format(dFrame, digits = 3), file=paste(ds, "MetaVs",label,"ShortLinearMod.txt", sep = ''), row.names=FALSE, sep="\t")
  
  pubDf = reshape(dFrame, idvar = 'indexName', timevar = 'metaName', direction="wide", sep="", drop = "index")
  write.table(format(pubDf, digits = 3), file=paste(ds, "MetaVs",label,"ReformatShortLinearMod.txt", sep = ''), row.names=FALSE, sep="\t")
    
  #print(pubDf)
  
  pdf(paste(ds,"MetaVs",label,"LinearMod.pdf", sep = ""))
  
  
  for( i in 1:nrow(dFrame)){
  	aTitle <- paste(label, ': ', dFrame$indexName[i], " vs ",  dFrame$metaName[i], "\nAdj Pval: ", formatC(dFrame$metaPvalAdj[i]), sep = '')
  	
    par(bty = 'l', 
    mar = c(5, 4, 5, 2) + 0.1,
    # cex = 1.3,
    # cex.lab = 1.5,
    cex.axis = 1.45
    )
    plot( as.factor(as.character(metadata[as.character(dFrame$metaName[i]),])),
      unlist(testingData[dFrame$index[i],]),
      main = aTitle,
      xlab = as.character(dFrame$metaName[i]),
      ylab = paste(as.character(dFrame$indexName[i])),
      col = brewer.pal(length(metadata[as.character(dFrame$metaName[i]),]),"Accent")
      )
  }
  dev.off()
  
  print('lmMetadata complete')
}#end of lmMetadata
bioActiveArr= c('Kavain', 'Dihydromethysticin', 'Desmethoxyyangonin')
mainBioActive = metabolites[bioActiveArr,]
lmMetadata(mainBioActive, 'BioActiveMetabolitesOnly')
```

Compare each metabolite to each metadata.
  Expected output:
    Table1: 
      Columns: anova adjusted pval (sorted), metadata name, metabolite name
    Table2: 
      Names and rows of metadata and metabolites for failed attempts,
    Table3:
      Posthoc pvalues for 'plant part'
    Table4 and Table5: <- this turned out too hard for the model
      Post hoc pvalues for 'tissue' and 'plant part'
        Table5 Columns:
          Metabolite | lateral_root - crown_root | stem - crown_root | stem - lateral_root
    Box plots for each metabolite and each metadata
```{r}
lmeMetadata = function(testingData, label){
  if(!require(nlme)){install.packages("nlme")}
  library("nlme")
  
  metaboliteNames = c()
  metadataCatagories = c()
  pvals = c()
  
  #vessel to hold the failures for future reference
  failedLme <<- data.frame(#'<<-' gives global scope
    metaboliteName = character(0), 
    metaboliteRowNum = numeric(0), 
    metadataName = character(0),
    metadataRowNum = numeric(0)
    )
  
  for (i1 in 1:nrow(testingData)){
    #print(paste('i1:',i1))
    for (i2 in 1:nrow(metadata)){
      #print(paste('i2:',i2))
      if (row.names(metadata)[i2] != 'plant number'){
        tryCatch(
          {
          metabo = unlist(testingData[i1,])
          metad = as.factor(unlist(metadata[i2,]))
          pn = unlist(metadata['plant number',])
          myD = data.frame(metabo, metad, pn)
          myLme = lme(metabo ~ metad, 
            method= "REML", 
            random = ~1 | pn, 
            na.action = 'na.omit',
            data = myD
            )
          myAnova = anova(myLme)
          myPval = myAnova$"p-value"[2]
          metaboliteNames = c(metaboliteNames, row.names(testingData)[i1])
          metadataCatagories = c(metadataCatagories, row.names(metadata)[i2])
          pvals =  c(pvals, myPval)
          },
          error=function(cond) {
            #metaboliteName, metaboliteRowNum, metadataName, metadataRowNum
            myRow = list(row.names(testingData)[i1], i1, row.names(metadata)[i2], i2)
            names(myRow) = c('metaboliteName', 'metaboliteRowNum', 'metadataName', 'metadataRowNum')
           # print(metabo)
            failedLme <<- rbind(failedLme, data.frame(myRow))
            #print('an error is thrown')
            message(cond)
            # Choose a return value in case of error
            # return(NA)
          },
          warning=function(cond) {
            print('a warning is thrown')
            myRow = list(row.names(testingData)[i1], i1, row.names(metadata)[i2], i2)
            names(myRow) = c('metaboliteName', 'metaboliteRowNum', 'metadataName', 'metadataRowNum')
            #print(names(myRow))
            failedLme = rbind(failedLme, data.frame(myRow))
            #print('failedLme warning')
            message(cond)
            # Choose a return value in case of warning
            #return(NULL)
          }
        )
      }
    }
  }
  
  pvalAdj = p.adjust(pvals, method = "BH")
  
  dFrame <- data.frame(pvals,pvalAdj,metaboliteNames,metadataCatagories)
  dFrame <- dFrame[order(dFrame$pvalAdj),]
  
  write.table(dFrame, file=paste(ds, "MetaVs", label, "Lme.txt", sep = ''), row.names=FALSE, sep="\t")
  
  write.table(failedLme, file=paste(ds, "FailedMetaVs", label, "Lme.txt", sep = ''), row.names=FALSE, sep="\t")
  
  pubDf = reshape(dFrame, idvar = 'metaboliteNames', timevar = 'metadataCatagories', direction="wide", sep="", drop = 'pvals')
  write.table(format(pubDf, digits = 3), file=paste(ds, "MetaVs",label,"ReformatShortLme.txt", sep = ''), row.names=FALSE, sep="\t")
  
  pdf(paste(ds, "HistogramMetaVs", label, ".pdf", sep=""))
  hist(pvals, 
    main = paste('Histogram of lme pvalues for ', label),
    xlab = 'Uncorrected pvalues',
    breaks = 50
  )
  dev.off()
  
  pdf(paste(ds, "MetaVs", label, "Lme.pdf", sep=""))
  par(bty = 'l', 
    mar = c(5, 4, 5, 2) + 0.1)
  for( i in 1:nrow(dFrame)){
  	aTitle <- paste(  dFrame$metaboliteNames[i], "vs",  dFrame$metadataCatagories[i], "\nAdjusted Pvalue=", dFrame$pvalAdj[i])
  
  	plot( as.factor(as.character(metadata[as.character(dFrame$metadataCatagories[i]),])),
  	  unlist(testingData[as.character(dFrame$metaboliteNames[i]),]),
  	  main = aTitle,
  	  xlab = as.character(dFrame$metadataCatagories[i]),
  	  ylab = as.character(dFrame$metaboliteNames[i]),
  	  col = as.factor(as.character(metadata[as.character(dFrame$metadataCatagories[i]),]))
  	  )
  }
  dev.off()
  print('successfull run of lmeMetadata')
}#end function
```


```{r}
lmeMetadata(metabolites, 'Metabolites')
```

```{r}
lmeMetadata(t(myPCA), 'PCA')
```
```{r}
myPcaT = data.frame(t(myPCA))
lmMetadata(myPcaT, 'Pca')
```

```{r}
lmMetadata(myPcaT[c(1,2,3),], "PcaTop3")
```

```{r}
lmeMetadata(myPcaT[c(1,2,3),], "PcaTop3")
```


```{r}
lmMetadata(metabolites, 'Metabolites')
```

Pairwise tests (code from Dr. Fodor https://github.com/afodor/metagenomicsTools/blob/master/src/scripts/aaronMetabolites/quickPairwise.txt)
```{r}

```

```{r}
expGroups = unique(unlist(metadata['tissue',]))
numGroups = length(expGroups)
iGroups <- vector()
jGroup <- vector()
index <- 1
pValues <- vector()
metabloliteName <- vector()
for( m in 1:nrow(metabolites) )
{	
	for ( i in 1:(numGroups-1) ) 
	{
		iData <- metabolites[m, which(metadata['tissue', ] == expGroups[i])]
		
		for( j in i:numGroups) 
		{
			jData <- metabolites[m,which(metadata['tissue', ] == expGroups[j])]
			if( sum(!is.na(iData)) >1 & sum(!is.na(jData)) >1   )
			{
				iGroups[index]  <- expGroups[i]
				jGroup[index] <-expGroups[j]
				pValues[index] <- t.test(iData, jData)$p.value
				metabloliteName[index] <- row.names(metabolites)[m]
				index = index + 1
				#print( paste(index, " " , metabloliteName, "\n"))
			}
		}
	}
}
myFrame <-data.frame(metabloliteName,iGroups,jGroup, pValues)
myFrame$pvalAdj = p.adjust(pValues, method="BH")
myFrame = myFrame[order(myFrame$pvalAdj),]
write.table(myFrame, file=paste(ds, 'PairwiseTtest.txt', sep = ''), row.names=FALSE, sep="\t")
sum(p.adjust(pValues, method="BH") < .1)
hist(pValues, 
  breaks = 50,
  xlab = 'Uncorrected pvalues',
  main = 'Pairwise t-tests'
  )
```

For publication purposes, I need to get just the metabolites of interest:
  Kavain
  Dihydromethysticin
  Desmethoxyyangonin
```{r}
#there will be two Dihydromethysticin in lc neg data
bioActiveArr= c('Kavain', 'Dihydromethysticin', 'Desmethoxyyangonin')
mainBioActive = metabolites[bioActiveArr,]
lmeMetadata(mainBioActive, 'BioActiveMetabolitesOnly')
```

```{r}
lmMetadata(mainBioActive, 'BioActiveMetabolitesOnly')
```


```{r}
lmMetadata(mainBioActive, 'BioActive')
```


```{r}
expGroups = unique(unlist(metadata['tissue',]))
numGroups = length(expGroups)
iGroups <- vector()
jGroup <- vector()
index <- 1
pValues <- vector()
metabloliteName <- vector()
for( m in 1:nrow(mainBioActive) )
{	
	for ( i in 1:(numGroups-1) ) 
	{
		iData <- mainBioActive[m, which(metadata['tissue', ] == expGroups[i])]
		
		for( j in i:numGroups) 
		{
			jData <- mainBioActive[m,which(metadata['tissue', ] == expGroups[j])]
			if( sum(!is.na(iData)) >1 & sum(!is.na(jData)) >1   )
			{
				iGroups[index]  <- expGroups[i]
				jGroup[index] <-expGroups[j]
				pValues[index] <- t.test(iData, jData)$p.value
				metabloliteName[index] <- row.names(mainBioActive)[m]
				index = index + 1
				#print( paste(index, " " , metabloliteName, "\n"))
			}
		}
	}
}
myFrame <-data.frame(metabloliteName,iGroups,jGroup, pValues)
myFrame$pvalAdj = p.adjust(pValues, method="BH")
myFrame = myFrame[order(myFrame$pvalAdj),]
write.table(myFrame, file=paste(ds, 'MainBioactive', 'PairwiseTtest.txt', sep = ''), row.names=FALSE, sep="\t")
sum(p.adjust(pValues, method="BH") < .1)
hist(pValues, 
  breaks = 50,
  xlab = 'Uncorrected pvalues',
  main = 'Pairwise t-tests'
  )
```


```{r}
#significant metabolites from the GC data lme
sigMetabols = "Tetrahydro-11-methoxyiangonin
Tetrahydro-11-methoxyiangonin
Tetrahydro-11-methoxyiangonin
Hedycaryol
Kavain
Kavain
Desmethoxyyangonin
Dihydromethysticin
Kavain
Desmethoxyyangonin
Yangonin
5,7-Dimethoxyflavanone
Hispidin
Hedycaryol
Dihydromethysticin
Pipermethystine
Flavokavain C
Pinostrobin
Pyrrolidine, 1-(m-methoxycinnamoyl)-
Flavokavain C
Pinostrobin
5,7-Dimethoxyflavanone
Hedycaryol
Bornyl cinnamate
5-Hydroxy-4',7-dimethoxyflavanone
Flavokavain B
Flavokavain B
5,7-Dimethoxyflavanone
5-Hydroxy-4',7-dimethoxyflavanone
Flavokavain C
Pinostrobin
Hispidin
Flavokavain B
Yangonin
Pyrrolidine, 1-(m-methoxycinnamoyl)-
Bornyl cinnamate
Bornyl cinnamate
Desmethoxyyangonin
Dihydromethysticin
Dihydro-5,6-dehydrokawain
Dihydro-5,6-dehydrokawain
Dihydroalatamide
Dihydro-5,6-dehydrokawain
5-Hydroxy-4',7-dimethoxyflavanone
Hispidin
Hydrocinnamic acid
Pyrrolidine, 1-(m-methoxycinnamoyl)-
Yangonin
Linoelaidic acid
Benzenepropanal
13-Tetradece-11-yn-1-ol
Linoelaidic acid
Hydrocinnamic acid
Palmitic Acid
13-Tetradece-11-yn-1-ol
13-Tetradece-11-yn-1-ol
Linoelaidic acid
Palmitic Acid
Benzenepropanal
Flavokawain A
Palmitic Acid
Hydrocinnamic acid
δ-Cadinol
Pipermethystine
δ-Cadinol
Flavokawain A"

sigMetabols = unlist(strsplit(sigMetabols,"\n"))

print(unique(sigMetabols))
```

```{r}
allMetabols = "Tetrahydro-11-methoxyiangonin
Tetrahydro-11-methoxyiangonin
Tetrahydro-11-methoxyiangonin
Hedycaryol
Kavain
Kavain
Desmethoxyyangonin
Dihydromethysticin
Kavain
Desmethoxyyangonin
Yangonin
5,7-Dimethoxyflavanone
Hispidin
Hedycaryol
Dihydromethysticin
Pipermethystine
Flavokavain C
Pinostrobin
Pyrrolidine, 1-(m-methoxycinnamoyl)-
Flavokavain C
Pinostrobin
5,7-Dimethoxyflavanone
Hedycaryol
Bornyl cinnamate
5-Hydroxy-4',7-dimethoxyflavanone
Flavokavain B
Flavokavain B
5,7-Dimethoxyflavanone
5-Hydroxy-4',7-dimethoxyflavanone
Flavokavain C
Pinostrobin
Hispidin
Flavokavain B
Yangonin
Pyrrolidine, 1-(m-methoxycinnamoyl)-
Bornyl cinnamate
Bornyl cinnamate
Desmethoxyyangonin
Dihydromethysticin
Dihydro-5,6-dehydrokawain
Dihydro-5,6-dehydrokawain
Dihydroalatamide
Dihydro-5,6-dehydrokawain
5-Hydroxy-4',7-dimethoxyflavanone
Hispidin
Hydrocinnamic acid
Pyrrolidine, 1-(m-methoxycinnamoyl)-
Yangonin
Linoelaidic acid
Benzenepropanal
13-Tetradece-11-yn-1-ol
Linoelaidic acid
Hydrocinnamic acid
Palmitic Acid
13-Tetradece-11-yn-1-ol
13-Tetradece-11-yn-1-ol
Linoelaidic acid
Palmitic Acid
Benzenepropanal
Flavokawain A
Palmitic Acid
Hydrocinnamic acid
δ-Cadinol
Pipermethystine
δ-Cadinol
Flavokawain A
α-epi-7-epi-5-Eudesmol
Squalene
Pipermethystine
Dihydrokavain
(E)-N-Cinnamoylpiperidine
Flavokawain A
Dihydroalatamide
α-epi-7-epi-5-Eudesmol
Dihydrokavain
(E)-N-Cinnamoylpiperidine
Dihydrokavain
Dihydroalatamide
Squalene
Squalene
1,10-Undecadiene"

allMetabols = unlist(strsplit(allMetabols,"\n"))

print(unique(allMetabols))
```
```{r}
#Values in sigMetabols that are not in allMetabols
notSig = allMetabols[!match(allMetabols, sigMetabols, nomatch = F)]
print(unique(notSig))
```

We want to test for covariance among the tissue types.  Specifically, we want to know if the "whole" tissues covary with the "peels" and "no peel" sections.

First step is to put the data in a long format so that 
```{r}
metabo_long1 = metabolites[,seq(1,21,3)]
metabo_long2 = metabolites[,seq(2,21,3)]
row.names(metabo_long2) = paste(row.names(metabo_long2), '1')
metabo_long3 = metabolites[,seq(3,21,3)]
row.names(metabo_long3) = paste(row.names(metabo_long2), '2')
names(metabo_long2) = names(metabo_long1)
names(metabo_long3) = names(metabo_long1)

metabo_long = rbind(metabo_long1,metabo_long2,metabo_long3)
```

Now we need to make a correlation matrix with metabo_long.
```{r}
metabolite_cor = cor(metabo_long, use='na.or.complete')
```


